### ISSUE TITLE
Migrate to new JavaDoc jdk.javadoc.doclet API

### ISSUE THREAD

**Issue by arolfes** [Taskana/taskana#2152]
```
Constructing Javadoc information...
Warning:  Javadoc Warnings
Warning:  javadoc: warning - The old Doclet and Taglet APIs in the packages
Warning:  com.sun.javadoc, com.sun.tools.doclets and their implementations
Warning:  are planned to be removed in a future JDK release. These
Warning:  components have been superseded by the new APIs in jdk.javadoc.doclet.
Warning:  Users are strongly recommended to migrate to the new APIs.
```

**Comment by arolfes** [comment]
This is currently blocked by:  https://github.com/ScaCap/spring-auto-restdocs/issues/412
